app/components/form/fields/SshKeysField.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useState } from 'react'
+import { useController, type Control } from 'react-hook-form'
+
+import { usePrefetchedApiQuery } from '@oxide/api'
+import {
+  Button,
+  Checkbox,
+  Divider,
+  EmptyMessage,
+  FieldLabel,
+  Key16Icon,
+  Message,
+  TextInputHint,
+} from '@oxide/ui'
+
+import type { InstanceCreateInput } from 'app/forms/instance-create'
+import { CreateSSHKeySideModalForm } from 'app/forms/ssh-key-create'
+
+import { CheckboxField } from '..'
+import { ErrorMessage } from './ErrorMessage'
+
+const MAX_KEYS_PER_INSTANCE = 8
+
+const CloudInitMessage = () => (
+  <Message
+    variant="notice"
+    className="mt-4"
+    content={
+      <>
+        If your image supports the cidata volume and{' '}
+        <a
+          target="_blank"
+          href="https://cloudinit.readthedocs.io/en/latest/"
+          rel="noreferrer"
+        >
+          cloud-init
+        </a>
+        , the keys above will be added to your instance. Keys are added when the instance is
+        created and are not updated after instance launch.
+      </>
+    }
+  />
+)
+
+export function SshKeysField({ control }: { control: Control<InstanceCreateInput> }) {
+  const keys = usePrefetchedApiQuery('currentUserSshKeyList', {}).data?.items || []
+  const [showAddSshKey, setShowAddSshKey] = useState(false)
+
+  const {
+    field: { value, onChange },
+    fieldState: { error },
+  } = useController({
+    control,
+    name: 'sshKeys',
+    rules: {
+      validate(keys) {
+        if (keys.length > MAX_KEYS_PER_INSTANCE) {
+          return `An instance supports a maximum of ${MAX_KEYS_PER_INSTANCE} SSH keys`
+        }
+      },
+    },
+  })
```

Comment:
It turns out this does work. I had forgotten that we changed all form validation to happen on submit, so this will not validate the number of keys until the user clicks submit. It's not ideal as a UX, but it is consistent with everything else. After that initial validation on submit, I think it validates on change, so that is a nice experience.


https://github.com/oxidecomputer/console/assets/3612203/d6f66745-9417-4610-be87-4f23eae0e9c9

Reply:
Now that this limit has increased to 100, I think it'll be a relatively uncommon occurrence.

Reply:
We'll need a nicer UI for longer lists here eventually I think. Looks good for now though!